src/mds/LogSegment.h

```diff
@@ -52,7 +52,10 @@ class LogSegment {
   {}
 
   void try_to_expire(MDSRank *mds, MDSGatherBuilder &gather_bld, int op_prio);
-
+  void purge_inode_finish_waiting(inodeno_t ino, int result = 0);
+  void add_waiter_for_purge_inode(inodeno_t ino, MDSContext* c){
+    purge_inodes_waiters.insert(std::pair<inodeno_t, MDSContext*>(ino, c));
+  }
```

Comment:
LogSegment::try_to_expire() waits for all inodes get purged. no need to add context for individual inode

Reply:
updated